force-app/main/default/lwc/bulkServiceDeliveryUI/bulkServiceDeliveryUI.js

```diff
@@ -155,23 +157,39 @@ export default class BulkServiceDeliveryUI extends NavigationMixin(LightningElem
     handleRowError(event) {
         let errorIndex = event.detail.index;
         this.errors[errorIndex] = "error";
+        this.isAddEntryDisabled = false;
         this.renderErrors();
     }
 
-    handleClearError(event) {
-        let errorIndex = event.detail;
-        let result = this.serviceDeliveries.find(({ index }) => index === errorIndex);
-        result.hasSaved = true;
-        this.handleDeleteError(errorIndex);
-    }
-
     handleDeleteError(index) {
         if (this.errors[index]) {
             delete this.errors[index];
             this.renderErrors();
         }
     }
 
+    handleSubmit() {
+        this.isAddEntryDisabled = true;
+        this.isDoneDisabled = true;
+    }
+
+    handleSuccess(event) {
+        this.isAddEntryDisabled = false;
+
+        //If previously there was an error and then the record saved successfully the below piece of code removes that error
+        //by calling the handleDeleteError method
+        let errorIndex = event.target.index;
+        let result = this.serviceDeliveries.find(({ index }) => index === errorIndex);
+        result.hasSaved = true;
+        this.handleDeleteError(errorIndex);
+
+        if (Object.keys(this.errors).length > 0) {
```

Comment:
Could this check be moved into render errors?

Reply:
@jjbennett , If we move this check to `renderErrors` it will be only called if there is an error and if there is no error the `Done` button will be disabled. If we enable the button in `handleSuccess` even if there is an error on any row the `done` button will be enabled since `handleSuccess` is called for every row.  So I think we should leave the check here unless I am missing something.

Reply:
It looks like handleDeleteError calls renderErrors so on 184 we'd also be in effect calling renderErrors.